README.md

```diff
@@ -245,6 +245,131 @@ in the correct format as Protobuf encoded binary data.
 should you want to use the `BigQuerySchema` instead.
 
+### Batch client
+
+The batch client can be used if you want to upload a big dataset of data to bigquery without any additional cost.
+
+Here follows an example on how you can create such a batch API driven BigQuery client.
+
+```go
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+
+    "github.com/OTA-Insight/bqwriter"
+
+    "cloud.google.com/go/bigquery"
+)
+
+// TODO: use more specific context
+ctx := context.Background()
+
+// create a BQ (stream) writer thread-safe client,
+bqWriter, err := bqwriter.NewStreamer(
+    ctx,
+    "my-gcloud-project",
+    "my-bq-dataset",
+    "my-bq-table",
+    &bqwriter.StreamerConfig{
+        // use 5 background worker threads
+        WorkerCount: 5,
+        // create the client with the JSON format and using autodetect
+        BatchClient: &bqwriter.BatchClientConfig{
+            AutoDetect: true,
+			SourceFormat: bigquery.JSON
+        },
+    },
+)
+)
+if err != nil {
+    // TODO: handle error gracefully
+    panic(err)
+}
+// do not forget to close, to close all background resources opened
+// when creating the BQ (stream) writer client
+defer bqWriter.Close()
+
+// Create some data, make sure your data implements the io.Reader interface so it can be used!
+type person struct {
+    ID int `json:"id"`
+    Name int `json:"name"`
+}
+
+data := []person{
+    {
+        ID: 1,
+        Name: "John"
+    },
+    {
+        ID: 2,
+        Name: "Jane"
+    }
+}
+
+// Convert the data into a reader.
+var buffer bytes.Buffer
+for r := range data {
+    body, _ := json.Marshal(r)
+    buffer.Write(body)
+    buffer.WriteString("\n") // If the format is JSON, make sure every row is on a newline.
+}
+
+// Write the data to bigquery.
+_, err := bqWriter.Put(bytes.NewReader(buffer.Bytes()))
+if err != nil {
+    // TODO: handle error gracefully
+    panic(err)
+}
+```
+
+You must define the `BatchClientConfig`, as demonstrated in previous example,
+in order to create a Batch client.
+Note that you cannot create a blank `BatchClientConfig` or any kind of default,
+as you are required to configure it with at least a `SourceFormat`.
+
+
+**BatchClientConfig options**
+
+- `BigQuerySchema` BigQuerySchema can be used in order to use a data encoder for the batchClient
+  based on a dynamically defined BigQuery schema in order to be able to encode any struct,
+  JsonMarshaler, Json-encoded byte slice, Stringer (text proto) or string (also text proto)
+  as a valid protobuf message based on the given BigQuery Schema. 
+
+  This config is only required if autoDetect is false.
+
+- `SourceFormat` is used to define the format that the data is that we will send.
+  Possible options are:
+  - bigquery.CSV
+  - bigquery.Avro
+  - bigquery.JSON
+  - bigquery.Parquet
+  - bigquery.ORC
+
+- `AutoDetect` can be used so that the client will determine the data format from the data
```

Comment:
I wonder if this cannot be left out, and simply use that if `BigQuerySchema` is true, as it anyway is a bit of an exclusive option, it's either one or the other.

Reply:
I've left it out as its not really an option that should be passed, but rather be set dynamically based on the `SourceFormat`